### PR TITLE
getting_started/build-guide: update Build SOF with zephyr

### PR DESCRIPTION
**Pull request by aborisovich**
Introduction of west tool to SOF project in PR https://github.com/thesofproject/sof/pull/6005 required to update
build guide on how to use python convenience script and west tool.

Signed-off-by: Andrey Borisovich <andrey.borisovich@intel.com>

### CODE REVIEW

File: getting_started/build-guide/build-with-zephyr.rst [hunk 1]
```diff
@@ -12,108 +12,217 @@ This guide describes how to build and run |SOF| as a Zephyr application.
...
-      ls zephyr/boards/xtensa/
-      west build --board intel_adsp_cavs25 ./zephyr/samples/subsys/audio/sof/
+      ./sof/scripts/xtensa-build-zephyr.py -u
+
+   - Initialize west to ``./sof/west.yml`` manifest is not already initialized.
+   - Clone and checkout projects to revisions defined in ``./sof/west.yml`` file.
+   - Skip building firmware image.
+
+Output directory
+   For convenience, the ``xtensa-build-zephyr.sh`` script copies all
```
Comment (kv2019i) [inline getting_started/build-guide/build-with-zephyr.rst:137]: "xtensa-build-zephyr.py"

File: getting_started/build-guide/build-with-zephyr.rst [hunk 2]
```diff
@@ -12,108 +12,217 @@ This guide describes how to build and run |SOF| as a Zephyr application.
...
+
+      cmake -B ./build-rimage -S ./sof/rimage
+      cmake --build ./build-rimage
+
+#. Sign firmware using rimage tool and a private key
 
    .. code-block:: bash
 
-      $ tree build-sof-staging/
+      west sign --build-dir ./build-tgl -t rimage --tool-path ./build-rimage/rimage --tool-data ./sof/rimage/config -- -k ./sof/keys/otc_private_key_3k.pem
```
Comment (aiChaoSONG) [inline getting_started/build-guide/build-with-zephyr.rst:205]: do we want to mention the -f option? or the version in manifest will be 0.0.0, which is not correct.

Comment (aborisovich) [reply getting_started/build-guide/build-with-zephyr.rst:205]: -f option is for `-f, --force           ignore any errors and try to proceed` according to `west sign --help` .
How is that related to versioning?

**Comment by aborisovich** [comment]
Push update:

- Added Troubleshooting section describing issue when west tool version is not sufficient to process sof/west.yml manifest.
As @mmaka1 noticed west tool 0.12.0 returns not very user-friendly Python exception.

**Review by lgirdwood — APPROVED** [review]
@deb-intel are you ok to fixup any grammar after merge ?

**Comment by aborisovich** [comment]
Hold on one more fix!

**Comment by aborisovich** [comment]
Now it's ok. Ready to merge.

**Comment by lgirdwood** [comment]
We can fix up any grammar later - lets get the devs using this.

**Comment by deb-intel** [comment]
@aborisovich I'm making language and style updates to this doc. On line 20, you state that the easiest way to build Z is to use its recommended toolchain. The link you provide does not exist and the section on using toolchains is not easily apparent (`Install a Toolchain <https://docs.zephyrproject.org/latest/getting_started/index.html#install-a-toolchain>`_). Do you mean to state that the easiest installation method is to use the Zephyr SDK which contains toolchains <https://docs.zephyrproject.org/latest/develop/getting_started/#install-zephyr-sdk>? Please advise, thanks!

**Comment by aborisovich** [comment]
> @aborisovich I'm making language and style updates to this doc. On line 20, you state that the easiest way to build Z is to use its recommended toolchain. The link you provide does not exist and the section on using toolchains is not easily apparent (`Install a Toolchain <https://docs.zephyrproject.org/latest/getting_started/index.html#install-a-toolchain>`_). Do you mean to state that the easiest installation method is to use the Zephyr SDK which contains toolchains https://docs.zephyrproject.org/latest/develop/getting_started/#install-zephyr-sdk? Please advise, thanks!

Yes, I meant Zephyr SDK.
In fact this part is a left over from the old version of the guide.

**Pull request merged**